pkg/readiness/ready_tracker_unit_test.go

```diff
@@ -162,3 +214,60 @@ func Test_ReadyTracker_TryCancelTemplate_Retries(t *testing.T) {
 		t.Fatal("tracker with 0 retries and cancellation should be satisfied")
 	}
 }
+
+func Test_Tracker_TryCancelData(t *testing.T) {
+	l := lister{syncsets: true}
+	for _, tt := range []struct {
+		name      string
+		objDataFn func() objData
+	}{
+		{name: "no retries"},
+		{name: "with retries", objDataFn: func() objData {
+			return objData{retries: 2}
+		}},
+	} {
+		rt := newTracker(l, false, false, false, tt.objDataFn)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		var runErr error
+		runWg := sync.WaitGroup{}
+		runWg.Add(1)
+		go func() {
+			runErr = rt.Run(ctx)
+			runWg.Done()
+		}()
```

Comment:
Please add a comment saying that we need this wait group so we can wait for the readytracker to stop before completing the test.